databallpy/features/covered_distance.py

```diff
@@ -0,0 +1,473 @@
+        raise e
+
+
+#%% 3-7-2024
+def _parse_intervals(intervals):
+    if not all(isinstance(element, (int, float)) for element in intervals):
+        raise TypeError("All elements in the tuple must be integers or floats")
+    
+    if len(intervals) % 2 != 0:
+        raise ValueError("Intervals must contain an even number of elements.")
+    
+    return [(min(intervals[i], intervals[i+1]), max(intervals[i], intervals[i+1])) for i in range(0, len(intervals), 2)]
+
+def get_covered_distance(
+    tracking_data: pd.DataFrame,
+    player_ids: list[str],
+    framerate: int,
+    velocity_intervals: tuple[float, ...] | None = None,
+    acceleration_intervals: tuple[float, ...] | None = None
+    ) -> dict:
+    """Function that calculates the distance covered in specified velocity and acceleration intervals
+
+    Args:
+        tracking_data:      tracking data
+        player_ids:         list containing players. Example = ['home_8', 'away_3', 'home_16']
+        framerate:          sample frequency in Hertz
+        velocity_intervals: tuple that contains the velocity interval(s). For example [0 3] & [4 8]: (0,3,8,4)
+        acc_intervals:      tuple that contains the acceleration interval(s). For example [-3 0] & [3 5]: (-3,0,3,5)      
+
+    Returns:
+        dict: a dictionary with for every player the total distance covered and optionally the 
+        distance covered within the given velocity and/or acceleration threshold(s)
+    
+    Notes: 
+        The function requires the velocity for every player calculated with the add_velocity function. 
+        The acceleration for every player depends on the presence of acceleration intervals in the input
+    """
+
+    try:
+        # Check input types
+        # tracking_data
+        if not isinstance(tracking_data, pd.DataFrame):
+            raise TypeError(f"tracking data must be a pandas DataFrame, not a {type(tracking_data).__name__}")
+        
+        # player_ids
+        if not isinstance(player_ids, list):
+            raise TypeError(f"player_ids must be a list, not a {type(player_ids).__name__}")
+
+        if not all(isinstance(player, str) for player in player_ids):
+            raise TypeError("All elements in player_ids must be strings")
+        
+        # framerate
+        if not isinstance(framerate, int):
+            raise TypeError(f"framerate must be a int, not a {type(framerate).__name__}")
+
+        # check velocity
+        for player_id in player_ids:
+            if (
+                player_id + "_vx" not in tracking_data.columns
+                or player_id + "_vy" not in tracking_data.columns
+                or player_id + "_velocity" not in tracking_data.columns
+            ):
+                raise ValueError(
+                    f"Velocity was not found for {player_id} in the DataFrame. "
+                    " Please calculate velocity first using add_velocity() function."
+                )
```

Comment:
Add sanity checks in a different function that is called at the beginning, something like `_validate_inputs`